### ISSUE TITLE
Le_beullenst test has an inconsistent behavior

### ISSUE THREAD

**Issue by Dioprz** [Crypto-TII/CryptographicEstimators#152]
When executing one of the tests runned by `docker-pytest`, the command

`sage tests/test_le_beullens.sage`

Has a very inconsistent behaviour:

```shell
root@d541a72bc243:/home/cryptographic_estimators# sage tests/test_le_beullens.sage
Traceback (most recent call last):
  File "/home/cryptographic_estimators/tests/test_le_beullens.sage.py", line 55, in <module>
    test_beullens2()
  File "/home/cryptographic_estimators/tests/test_le_beullens.sage.py", line 50, in test_beullens2
    assert t2 - RANGES < t1 < t2 + RANGES
AssertionError
root@d541a72bc243:/home/cryptographic_estimators# sage tests/test_le_beullens.sage
Traceback (most recent call last):
  File "/home/cryptographic_estimators/tests/test_le_beullens.sage.py", line 55, in <module>
    test_beullens2()
  File "/home/cryptographic_estimators/tests/test_le_beullens.sage.py", line 50, in test_beullens2
    assert t2 - RANGES < t1 < t2 + RANGES
AssertionError
root@d541a72bc243:/home/cryptographic_estimators# sage tests/test_le_beullens.sage # No error from here
root@d541a72bc243:/home/cryptographic_estimators# sage tests/test_le_beullens.sage
root@d541a72bc243:/home/cryptographic_estimators# sage tests/test_le_beullens.sage
root@d541a72bc243:/home/cryptographic_estimators# sage tests/test_le_beullens.sage
```

This issue must be addressed in the next phase of the SageMath migration.

**Comment by FloydZ** [comment]
@Dioprz is this still an issue? And if so on which branch are you working. The problem is that the script by Beullens is faulty, hence the Estimator is computing something slightly different. Thats why we added this `ranges` parameter.

**Comment by Dioprz** [comment]
Got it. Just let me ask:

1. Do you know why errors stop after some executions? It feels like if something is being cacheed, but I'm not sure.
2. Anything we can do to fix/improve the reliability of the test?
3. Should/could the range be increased, so it works even from the first execution?

**Comment by Memphisd** [comment]
This issue is actually related to the probabilistic behavior of the computation of beullens running time. Precisely the function `median_size_of_random_orbit` in the `le_helper.py` file. It computes a value needed for the time complexity as the median of a 100 random samples. If this median, by chance is off from what one would usually expect the value of the time complexity slightly changes. I suggest we simply increase the tolerance to compensate for the variance in the median computation. I tested the execution with `ranges=1.2` and restricting to a single runtime computation for both estimators, i.e., removing the `min(...)`. This should also fix the related problem in PR #188 .

**Comment by Dioprz** [comment]
Solved during the Sage to Python migration with #188. Thanks again for your insights, @Memphisd.
